modules/tinymce/src/core/test/ts/browser/keyboard/SpaceKeyTest.ts

```diff
@@ -93,6 +93,14 @@ describe('browser.tinymce.core.keyboard.SpaceKeyTest', () => {
       TinyAssertions.assertContent(editor, '<p>s&nbsp;<span style="display: block;" contenteditable="false">a</span></p>');
     });
 
+    it('TINY-8588: Add one space just after a block', () => {
+      const editor = hook.editor();
+      editor.setContent('<p><span style="display: block;" contenteditable="false">a</span>s</p>');
+      TinySelections.setCursor(editor, [ 0, 1 ], 0);
+      editor.selection.setContent(' ');
```

Comment:
I must have missed this before, but why are we doing `editor.selection.setContent(' ')` here? Don't we want to be pressing the space key here instead since that is the bug and a more direct use case of `Nbsp.ts`?